parallel_gripper_controller/include/parallel_gripper_controller/parallel_gripper_action_controller_impl.hpp

```diff
@@ -237,27 +237,34 @@ void GripperActionController::check_for_success(
   {
     if (fabs(current_velocity) > params_.stall_velocity_threshold)
     {
-      last_movement_time_ = time;
+      last_movement_time_.try_set(time);
     }
-    else if ((time - last_movement_time_).seconds() > params_.stall_timeout)
+    else
     {
-      pre_alloc_result_->state.effort[0] = computed_command_;
-      pre_alloc_result_->state.position[0] = current_position;
-      pre_alloc_result_->reached_goal = false;
-      pre_alloc_result_->stalled = true;
-
-      if (params_.allow_stalling)
-      {
-        RCLCPP_DEBUG(get_node()->get_logger(), "Stall detected moving to goal. Returning success.");
-        active_goal->setSucceeded(pre_alloc_result_);
-      }
-      else
+      auto last_time_opt = last_movement_time_.try_get();
+      if (
+        last_time_opt.has_value() &&
+        (time - last_time_opt.value()).seconds() > params_.stall_timeout)
       {
-        RCLCPP_DEBUG(get_node()->get_logger(), "Stall detected moving to goal. Aborting action!");
-        active_goal->setAborted(pre_alloc_result_);
+        pre_alloc_result_->state.effort[0] = computed_command_;
+        pre_alloc_result_->state.position[0] = current_position;
+        pre_alloc_result_->reached_goal = false;
+        pre_alloc_result_->stalled = true;
+
+        if (params_.allow_stalling)
+        {
+          RCLCPP_DEBUG(
+            get_node()->get_logger(), "Stall detected moving to goal. Returning success.");
+          active_goal->setSucceeded(pre_alloc_result_);
+        }
+        else
+        {
+          RCLCPP_DEBUG(get_node()->get_logger(), "Stall detected moving to goal. Aborting action!");
+          active_goal->setAborted(pre_alloc_result_);
+        }
+        rt_active_goal_.set([](RealtimeGoalHandlePtr & stored_value)
```

Comment:
not related to the original PR, but rt_active_goal_ also uses non-RT get and set?

Reply:
I'm happy you brought this up. I think, `rt_active_goal_` should use `try_get()` and `try_set() `as well, I was not sure why it uses `set()` and `get()`.